commons/commons-serialization/src/main/kotlin/ac/uk/ebi/biostd/service/FileListSerializer.kt

```diff
@@ -37,6 +38,7 @@ internal class FileListSerializer(
             is FireBioFile -> TODO()
             is FireDirectoryBioFile -> TODO()
             is NfsBioFile -> bioFile.file
+            null -> throw FileNotFoundException(fileList)
```

Comment:
Ok, I see. The only bad thing about leaving the responsibility for the exception outside the sources is that it can be easily forgotten

Reply:
It can be forgotten in Java but not in Kotlin, as return type is nullable so compilter will force you. Code will not compile if you do not handle null case